### PR TITLE
Final submission for feature selection tutorial

### PR DESCRIPTION
**Pull request by bbeat2782**
This is my final submission of .md file that contains a tutorial that explains what a feature selection is, what methods there are, and how to implement those with sklearn. 

I would greatly appreciate it if you comment when you can't view what I submitted. 

----
:books: Documentation preview :books:: https://sklearn-evaluation--294.org.readthedocs.build/en/294/

### CODE REVIEW

**Comment by idomic** [comment]
This closes ##287

**Comment by idomic** [comment]
@gtduncan first step here is to make sure it shows as part of the docs and that the CI passes.

**Comment by neelasha23** [comment]
@bbeat2782 Please ensure the CI passes. Also, this tutorial doesn't show up int he docs preview. Please add it in the `_toc.yml` file

File: docs/optimization/feature_selection.md [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+```{code-cell} ipython3
+:tags: []
+
+%load_ext autoreload
+%autoreload 2
+```
+
+# Feature Selection
+
+In a dataset, not every feature might be useful. Some could be noise, and some could show a strong relationship with the target variable. Often, using all the features decreases a model's performance and increases training time. Therefore, it's best to select the best subset of features that are relevant to the target variable before we train a model. This process is called feature selection.
```
Comment (neelasha23) [inline docs/optimization/feature_selection.md:23]: `best to select the best subset` this is sounding redundant.
Briefly mention what are features vs target variables. mention overfitting here.

File: docs/optimization/feature_selection.md [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+In a dataset, not every feature might be useful. Some could be noise, and some could show a strong relationship with the target variable. Often, using all the features decreases a model's performance and increases training time. Therefore, it's best to select the best subset of features that are relevant to the target variable before we train a model. This process is called feature selection. 
+
+With supervised learning, feature selection has 3 main categories.
+
+#### 1. Filter method
+#### 2. Wrapper method
+#### 3. Embedded method
+
+In this tutorial, we will go over what those 3 categories are, what methods are under the 3 categories, and how to implement those with sklearn. So let's first bring in toy datasets from the sklearn package.
```
Comment (neelasha23) [inline docs/optimization/feature_selection.md:31]: just mention `datasets`. Remove `toy`

File: docs/optimization/feature_selection.md [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+3. Spearman's Rank Correlation
+4. ANOVA F-test
+5. Chi-Square Test
+6. Mutual Information
+
++++ {"tags": []}
+
+### 1.1. Quasi Constant
+
+A Quasi constant feature is a feature whose majority are the same values. If all or most of the values are the same, it won't be useful for prediction. So after you set a minimum threshold for variance, you remove features based on whether their variances exceed the threshold or not.
```
Comment (neelasha23) [inline docs/optimization/feature_selection.md:88]: reframe this sentence -> majority of the samples have same value of this feature

**Pull request closed without merging**